### PR TITLE
python3Packages.xmldiff: refactor

### PR DESCRIPTION
**Pull request by SFrijters**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/xmldiff/default.nix [hunk 1]
```diff
@@ -10,7 +10,7 @@
 buildPythonPackage rec {
   pname = "xmldiff";
   version = "2.6.3";
-  format = "setuptools";
+  pyformat = true;
```
Comment (pbsds) [inline pkgs/development/python-modules/xmldiff/default.nix:13]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/xmldiff/default.nix [hunk 2]
```diff
@@ -19,7 +19,7 @@ buildPythonPackage rec {
     hash = "sha256-GbAws/o30fC1xa2a2pBZiEw78sdRxd2PHrTtSc/j/GA=";
   };
```
Comment (pbsds) [inline pkgs/development/python-modules/xmldiff/default.nix:21]:
```suggestion
  build-system = [ setuptools ];
```

Comment (SFrijters) [reply pkgs/development/python-modules/xmldiff/default.nix:21]: It is (also) needed as a runtime dependency for the executables - should it be added both to `build-system` and `dependencies`? I have verified it needs the latter for sure.

Comment (SFrijters) [reply pkgs/development/python-modules/xmldiff/default.nix:21]: Done

**Comment by SFrijters** [comment]
Result of `nixpkgs-review pr 310097` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>fira-math</li>
    <li>python311Packages.fontmake</li>
    <li>python311Packages.fontmake.dist</li>
    <li>python312Packages.fontmake</li>
    <li>python312Packages.fontmake.dist</li>
  </ul>
</details>
<details>
  <summary>84 packages built:</summary>
  <ul>
    <li>atomic-operator</li>
    <li>atomic-operator.dist</li>
    <li>checkov</li>
    <li>checkov.dist</li>
    <li>crackmapexec</li>
    <li>crackmapexec.dist</li>
    <li>cyclonedx-python</li>
    <li>cyclonedx-python.dist</li>
    <li>fontbakery (python311Packages.fontbakery)</li>
    <li>fontbakery.dist (python311Packages.fontbakery.dist)</li>
    <li>knowsmore</li>
    <li>knowsmore.dist</li>
    <li>netexec</li>
    <li>netexec.dist</li>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
    <li>python311Packages.babelfont</li>
    <li>python311Packages.babelfont.dist</li>
    <li>python311Packages.collidoscope</li>
    <li>python311Packages.collidoscope.dist</li>
    <li>python311Packages.cyclonedx-python-lib</li>
    <li>python311Packages.cyclonedx-python-lib.dist</li>
    <li>python311Packages.fontfeatures</li>
    <li>python311Packages.fontfeatures.dist</li>
    <li>python311Packages.glyphsets</li>
    <li>python311Packages.glyphsets.dist</li>
    <li>python311Packages.glyphslib</li>
    <li>python311Packages.glyphslib.dist</li>
    <li>python311Packages.glyphtools</li>
    <li>python311Packages.glyphtools.dist</li>
    <li>python311Packages.psrpcore</li>
    <li>python311Packages.psrpcore.dist</li>
    <li>python311Packages.py-serializable</li>
    <li>python311Packages.py-serializable.dist</li>
    <li>python311Packages.pypsrp</li>
    <li>python311Packages.pypsrp.dist</li>
    <li>python311Packages.skytemple-dtef</li>
    <li>python311Packages.skytemple-dtef.dist</li>
    <li>python311Packages.skytemple-files</li>
    <li>python311Packages.skytemple-files.dist</li>
    <li>python311Packages.skytemple-ssb-debugger</li>
    <li>python311Packages.skytemple-ssb-debugger.dist</li>
    <li>python311Packages.sphinxcontrib-wavedrom</li>
    <li>python311Packages.sphinxcontrib-wavedrom.dist</li>
    <li>python311Packages.wavedrom</li>
    <li>python311Packages.wavedrom.dist</li>
    <li>python311Packages.xmldiff</li>
    <li>python311Packages.xmldiff.dist</li>
    <li>python312Packages.babelfont</li>
    <li>python312Packages.babelfont.dist</li>
    <li>python312Packages.collidoscope</li>
    <li>python312Packages.collidoscope.dist</li>
    <li>python312Packages.cyclonedx-python-lib</li>
    <li>python312Packages.cyclonedx-python-lib.dist</li>
    <li>python312Packages.fontbakery</li>
    <li>python312Packages.fontbakery.dist</li>
    <li>python312Packages.fontfeatures</li>
    <li>python312Packages.fontfeatures.dist</li>
    <li>python312Packages.glyphsets</li>
    <li>python312Packages.glyphsets.dist</li>
    <li>python312Packages.glyphslib</li>
    <li>python312Packages.glyphslib.dist</li>
    <li>python312Packages.glyphtools</li>
    <li>python312Packages.glyphtools.dist</li>
    <li>python312Packages.psrpcore</li>
    <li>python312Packages.psrpcore.dist</li>
    <li>python312Packages.py-serializable</li>
    <li>python312Packages.py-serializable.dist</li>
    <li>python312Packages.pypsrp</li>
    <li>python312Packages.pypsrp.dist</li>
    <li>python312Packages.skytemple-dtef</li>
    <li>python312Packages.skytemple-dtef.dist</li>
    <li>python312Packages.skytemple-files</li>
    <li>python312Packages.skytemple-files.dist</li>
    <li>python312Packages.skytemple-ssb-debugger</li>
    <li>python312Packages.skytemple-ssb-debugger.dist</li>
    <li>python312Packages.sphinxcontrib-wavedrom</li>
    <li>python312Packages.sphinxcontrib-wavedrom.dist</li>
    <li>python312Packages.wavedrom</li>
    <li>python312Packages.wavedrom.dist</li>
    <li>python312Packages.xmldiff</li>
    <li>python312Packages.xmldiff.dist</li>
    <li>skytemple</li>
    <li>skytemple.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/xmldiff/default.nix (hunk 2 above)
Comment (pbsds) [reply pkgs/development/python-modules/xmldiff/default.nix:21]: this is to properly support cross builds ;)

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 310097` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>fira-math</li>
    <li>python311Packages.fontmake</li>
    <li>python311Packages.fontmake.dist</li>
    <li>python312Packages.fontmake</li>
    <li>python312Packages.fontmake.dist</li>
  </ul>
</details>
<details>
  <summary>84 packages built:</summary>
  <ul>
    <li>atomic-operator</li>
    <li>atomic-operator.dist</li>
    <li>checkov</li>
    <li>checkov.dist</li>
    <li>crackmapexec</li>
    <li>crackmapexec.dist</li>
    <li>cyclonedx-python</li>
    <li>cyclonedx-python.dist</li>
    <li>fontbakery (python311Packages.fontbakery)</li>
    <li>fontbakery.dist (python311Packages.fontbakery.dist)</li>
    <li>knowsmore</li>
    <li>knowsmore.dist</li>
    <li>netexec</li>
    <li>netexec.dist</li>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
    <li>python311Packages.babelfont</li>
    <li>python311Packages.babelfont.dist</li>
    <li>python311Packages.collidoscope</li>
    <li>python311Packages.collidoscope.dist</li>
    <li>python311Packages.cyclonedx-python-lib</li>
    <li>python311Packages.cyclonedx-python-lib.dist</li>
    <li>python311Packages.fontfeatures</li>
    <li>python311Packages.fontfeatures.dist</li>
    <li>python311Packages.glyphsets</li>
    <li>python311Packages.glyphsets.dist</li>
    <li>python311Packages.glyphslib</li>
    <li>python311Packages.glyphslib.dist</li>
    <li>python311Packages.glyphtools</li>
    <li>python311Packages.glyphtools.dist</li>
    <li>python311Packages.psrpcore</li>
    <li>python311Packages.psrpcore.dist</li>
    <li>python311Packages.py-serializable</li>
    <li>python311Packages.py-serializable.dist</li>
    <li>python311Packages.pypsrp</li>
    <li>python311Packages.pypsrp.dist</li>
    <li>python311Packages.skytemple-dtef</li>
    <li>python311Packages.skytemple-dtef.dist</li>
    <li>python311Packages.skytemple-files</li>
    <li>python311Packages.skytemple-files.dist</li>
    <li>python311Packages.skytemple-ssb-debugger</li>
    <li>python311Packages.skytemple-ssb-debugger.dist</li>
    <li>python311Packages.sphinxcontrib-wavedrom</li>
    <li>python311Packages.sphinxcontrib-wavedrom.dist</li>
    <li>python311Packages.wavedrom</li>
    <li>python311Packages.wavedrom.dist</li>
    <li>python311Packages.xmldiff</li>
    <li>python311Packages.xmldiff.dist</li>
    <li>python312Packages.babelfont</li>
    <li>python312Packages.babelfont.dist</li>
    <li>python312Packages.collidoscope</li>
    <li>python312Packages.collidoscope.dist</li>
    <li>python312Packages.cyclonedx-python-lib</li>
    <li>python312Packages.cyclonedx-python-lib.dist</li>
    <li>python312Packages.fontbakery</li>
    <li>python312Packages.fontbakery.dist</li>
    <li>python312Packages.fontfeatures</li>
    <li>python312Packages.fontfeatures.dist</li>
    <li>python312Packages.glyphsets</li>
    <li>python312Packages.glyphsets.dist</li>
    <li>python312Packages.glyphslib</li>
    <li>python312Packages.glyphslib.dist</li>
    <li>python312Packages.glyphtools</li>
    <li>python312Packages.glyphtools.dist</li>
    <li>python312Packages.psrpcore</li>
    <li>python312Packages.psrpcore.dist</li>
    <li>python312Packages.py-serializable</li>
    <li>python312Packages.py-serializable.dist</li>
    <li>python312Packages.pypsrp</li>
    <li>python312Packages.pypsrp.dist</li>
    <li>python312Packages.skytemple-dtef</li>
    <li>python312Packages.skytemple-dtef.dist</li>
    <li>python312Packages.skytemple-files</li>
    <li>python312Packages.skytemple-files.dist</li>
    <li>python312Packages.skytemple-ssb-debugger</li>
    <li>python312Packages.skytemple-ssb-debugger.dist</li>
    <li>python312Packages.sphinxcontrib-wavedrom</li>
    <li>python312Packages.sphinxcontrib-wavedrom.dist</li>
    <li>python312Packages.wavedrom</li>
    <li>python312Packages.wavedrom.dist</li>
    <li>python312Packages.xmldiff</li>
    <li>python312Packages.xmldiff.dist</li>
    <li>skytemple</li>
    <li>skytemple.dist</li>
  </ul>
</details>

there is a bug with the `python3Packages` alias on ofborg, i'd like to check the other platforms

@ofborg build python311Packages.xmldiff python312Packages.xmldiff

**Pull request merged**